src/v/storage/log.h

```diff
@@ -96,8 +99,28 @@ class log {
     virtual ss::future<std::optional<timequery_result>>
       timequery(timequery_config) = 0;
 
+    // Prefer to use delta() or from/to_log_offset().
+    // TODO: remove direct access to the translator state and instead rely on
+    // the translation/delta interface.
+    virtual ss::lw_shared_ptr<const storage::offset_translator_state>
+    get_offset_translator_state() const = 0;
+
+    // Returns the offset delta for a given offset. This can be used for
+    // example to translate a Raft offset to a data offset.
+    virtual model::offset_delta delta(model::offset) const = 0;
```

Comment:
nit: `log.delta()` is much less descriptive than `offset_translator.delta()` so maybe a more specific name here would be better.